trts/src/lib.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2022 MobileCoin Foundation
+
+#![doc = include_str!("../README.md")]
+#![no_std]
+#![deny(missing_docs, missing_debug_implementations)]
+
+use core::{ffi::c_void, mem};
+use mc_sgx_trts_sys::{sgx_is_outside_enclave, sgx_is_within_enclave};
+
+/// Behavior for determining where memory values are at.
+pub trait EnclaveMemory<T> {
+    /// Is the item fully within the enclave's memory space?
+    ///
+    /// # Arguments
+    /// - `item` - The item to see if it's fully inside of the enclave's memory
+    ///   space
+    fn is_within_enclave(item: &T) -> bool;
+
+    /// Is the item fully outside of the enclave's memory space?
+    ///
+    /// # Arguments
+    /// - `item` - The item to see if it's fully outside of the enclave's memory
+    ///   space
+    fn is_outside_enclave(item: &T) -> bool;
+}
+
+impl<T: Sized> EnclaveMemory<T> for T {
+    fn is_within_enclave(item: &T) -> bool {
+        let start = item as *const _ as *const c_void;
+        let size = mem::size_of::<T>();
+        matches!(unsafe { sgx_is_within_enclave(start, size) }, 1)
```

Comment:
Why use `matches!` here?
```suggestion
        unsafe { sgx_is_within_enclave(start, size) } == 1
```

Reply:
yah, I wasn't too fond of `matches!` either.
I started with
```rust
        match unsafe { sgx_is_within_enclave(start, size) } {
            1 => true,
            _ => false,
        }
```
Then clippy gave me this
```shell
   |
31 | /         match unsafe { sgx_is_within_enclave(start, size) } {
32 | |             1 => true,
33 | |             _ => false,
34 | |         }
   | |_________^ help: try this: `matches!(unsafe { sgx_is_within_enclave(start, size) }, 1)`
   |
   = note: `#[warn(clippy::match_like_matches_macro)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_like_matches_macro
```
I've updated them to 
```rust
        let result = unsafe { sgx_is_within_enclave(start, size) };
        result == 1
```
one can't do 
```rust
unsafe { sgx_is_within_enclave(start, size) } == 1
```
but one can do
```rust
1 == unsafe { sgx_is_within_enclave(start, size) }
```

I felt the 2 lines was more intuitive and with rust's non mut by default one doesn't need to worry, as much, about C's `if(x = CONSTANT)` error.